### PR TITLE
base/v0_1: add translation

### PR DESCRIPTION
**Pull request by ajeddeloh**
Implement the actual meat of the translation. This doesn't have the
nicities of inlining files yet, but should have eveything else.

### CODE REVIEW

**Comment by ajeddeloh** [comment]
Ok, actually ready for review now. Does line/column reporting and has tests.

Doesn't have inline data for files yet, but that's sufficiently separable.

File: base/v0_1/translate.go [hunk 1]
```diff
@@ -1,13 +1,73 @@
...
+	tr.AddCustomTranslator(translateFile)
+	tr.AddCustomTranslator(translateDirectory)
+	tr.AddCustomTranslator(translateLink)
+	tr.Translate(&c, &ret)
+	return ret, nil
+}
+
+func translateIgnition(from Ignition) (to types.Ignition) {
+	tr := translate.NewTranslator()
+	to.Version = "3.0.0"
```
Comment (ashcrow) [inline base/v0_1/translate.go]: Should versions be constants referenced and reused?

File: base/v0_1/translate_test.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+	"reflect"
+	"testing"
+
+	"github.com/coreos/ignition/v2/config/util"
+	"github.com/coreos/ignition/v2/config/v3_0/types"
+)
+
+// Most of this is covered by the Ignition translator generic tests, so just test the custom bits
+
+func TestTranslateFile(t *testing.T) {
```
Comment (ashcrow) [inline base/v0_1/translate_test.go]: nit: I'm a fan of giving tests a simple document string to make it a bit easier to see at a glance what is meant to be tested.

File: config/common/common.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	Pretty bool
+	Strict bool
+}
+
+type Common struct {
+	Version string `yaml:"version"`
+	Variant string `yaml:"variant"`
+}
+
+// Misc helpers
```
Comment (ashcrow) [inline config/common/common.go]: nit: Doc

File: config/common/common.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+func Unmarshal(data []byte, to interface{}, strict bool) (tree.Node, error) {
+	dec := yaml.NewDecoder(bytes.NewReader(data))
+	dec.KnownFields(strict)
+	if err := dec.Decode(to); err != nil {
+		return nil, err
+	}
+	return vyaml.UnmarshalToContext(data)
+}
+
+func Marshal(from interface{}, pretty bool) ([]byte, error) {
```
Comment (ashcrow) [inline config/common/common.go]: nit: Doc

File: config/common/common.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+func camel(in string) string {
+	words := strings.Split(in, "_")
+	for i, word := range words[1:] {
+		words[i+1] = strings.Title(word)
+	}
+	return strings.Join(words, "")
+}
+
+func ToCamelCase(t tree.Node) tree.Node {
```
Comment (ashcrow) [inline config/common/common.go]: nit: Doc

**Comment by ajeddeloh** [comment]
@ashcrow pushed some fixup commits adding docs. LGTY?

**Pull request merged**